In [3]:
import pandas as pd

URL = "https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv"
df = pd.read_csv(URL)
df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


# Question
**1. Aperçu global : Combien de passagers ? Combien de NaN par colonne ?**

In [39]:
df.info()
df.isna().sum()

<class 'pandas.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    str    
 4   Sex          891 non-null    str    
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    str    
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    str    
 11  Embarked     889 non-null    str    
dtypes: float64(2), int64(5), str(5)
memory usage: 83.7 KB


PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

**2.Taux de survie global**

In [40]:
tx_survie = round(df["Survived"].mean()*100,2)
print("Le taux de survie global est " + str(tx_survie) + "%")

Le taux de survie global est 38.38%


**3.Taux de survie par classe**

In [41]:
df.groupby("Pclass")["Survived"].mean()

Pclass
1    0.629630
2    0.472826
3    0.242363
Name: Survived, dtype: float64

**4.Taux de survie par sexe**

In [57]:
df.groupby("Sex")["Survived"].mean()

Sex
female    0.742038
male      0.188908
Name: Survived, dtype: float64

**5.Taux de survie par classe et sexe**

In [58]:
df.groupby(["Pclass","Sex"])["Survived"].mean()

Pclass  Sex   
1       female    0.968085
        male      0.368852
2       female    0.921053
        male      0.157407
3       female    0.500000
        male      0.135447
Name: Survived, dtype: float64

**6.Tranches d'age**

In [59]:
def age_group(age):
    if pd.isna(age):
        return "inconnu"    
    elif age < 18:
        return "enfant"
    elif age < 60:
        return "adulte"
    else:
        return "senior"
df["Age_group"] = df["Age"].apply(age_group)
df.groupby("Age_group")["Survived"].mean()

Age_group
adulte     0.386087
enfant     0.539823
inconnu    0.293785
senior     0.269231
Name: Survived, dtype: float64

# Conclusion
Le jeu de données est sur le nauffrage du titanic, il comporte 12 coulonnes. Au total le jeu de données contient 891 passagers. Cependant plusieurs colonnes ont des valeurs null :
- 177 pour la colonne Age
- 687 pour la colonne Cabin
- 2 pour la colonne Embarked

Nous avons également regardé les taux de survie en fonctions de certaines informations.
Le taux de survie global du nauffrage est de 38.38% donc un peu plus d'une personne sur 3 a pu survire.
Le taux de survie en fonction des classes est ingégale. On peut voir que la classe 1 a un taux de survie de 62.9% soit presque 2/3 des passagers ont survie. La classe 2 a un taux de survie de 47.28% soit presque une personne sur 2. Ce qui est déjà moins que la classe 1 et la classe 3 a un taux de survie de 24.23% soit un peu moins d'une personne sur quatre. On peut donc déjà voir une disparité du taux de survie entre les classes.
Maintenant nous allons ajouté à l'analyse le sexe. Le taux de survie chez les hommes est de 18.89% soit moins d'un homme sur cinq ont survie et chez les femmes le taux de survie est de 74.2% soit près de trois femmes sur quatre ont survie.
En ajoutant les classes nous avons pour la classe 1 chez les hommes un taux de survie de  36.88% et chez les femmes un taux de survie de 96.8%. pour la classe 2 chez les hommes le taux de survie est de 15.74% et chez les femmes le taux est de 92.1%. Et pour la classe 3 chez les hommes le taux de survie est de 13.54% et chez les femmes le taux est de 50%. Nous pouvons donc voir que la classe joue un rôle majeur dans la survie au nauffrage mais que le sexe en est un encore plus élevé.
Pour regarder le taux de suvie en fonction de l'age j'ai créé une tranche d'age qui est :
- enfant: pour les moins de 18 ans
- adulte: pour les 18-60 ans 
- senior: pour les plus de 60 ans
- inconnus: pour le valeurs null

Le taux de survie pour les enfants est de 53.98%, pour les adultes il est de 38.6%, pour les seniors le taux de survie est de 26.92% et pour les inconnus il est de 29.37%